apiserver/facades/controller/caasunitprovisioner/provisioner.go

```diff
@@ -254,47 +259,25 @@ func (f *Facade) provisioningInfo(model Model, tagString string) (*params.Kubern
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	units, err := app.AllUnits()
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	// Can happen if scale is set to 0 in k8s, outside of Juju.
-	// In this case, there is no provisioning info to return.
-	if len(units) == 0 {
-		logger.Debugf("cannot provision application %q with no units", appTag.Id())
-		return nil, nil
-	}
 	modelConfig, err := model.ModelConfig()
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 
-	// Find the first alive unit which will be used to get filesystem info.
-	var aliveUnit Unit
-	for _, u := range units {
-		if u.Life() == state.Alive {
-			aliveUnit = u
-			break
-		}
-	}
-
 	controllerCfg, err := f.state.ControllerConfig()
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 
-	var filesystemParams []params.KubernetesFilesystemParams
-	if aliveUnit != nil {
-		filesystemParams, err = f.applicationFilesystemParams(controllerCfg, modelConfig, aliveUnit.UnitTag())
-		if err != nil {
-			return nil, errors.Trace(err)
-		}
+	filesystemParams, err := f.applicationFilesystemParams(app, controllerCfg, modelConfig)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
 
-		// The juju-storage-owner tag is set to the unit. We use it as a label on the CAAS volume.
-		// Since we used an arbitrary unit to get the info, reset the tag to the application.
-		for _, fsp := range filesystemParams {
-			fsp.Tags[tags.JujuStorageOwner] = appTag.Id()
-		}
+	// The juju-storage-owner tag is set to the application.
+	// We use it as a label on the CAAS volume.
+	for _, fsp := range filesystemParams {
+		fsp.Tags[tags.JujuStorageOwner] = appTag.Id()
```

Comment:
it's already done in `func filesystemParams`
```
filesystemTags[tags.JujuStorageOwner] = app.Name()
```

Reply:
ah yes, thanks